packages/nuxt3/src/app/entry.ts

```diff
@@ -43,19 +51,28 @@ if (process.client) {
 
     const nuxt = createNuxtApp({ vueApp })
 
-    await applyPlugins(nuxt, plugins)
-
-    await nuxt.hooks.callHook('app:created', vueApp)
-    await nuxt.hooks.callHook('app:beforeMount', vueApp)
-
     nuxt.hooks.hookOnce('app:suspense:resolve', () => {
       nuxt.isHydrating = false
     })
 
-    vueApp.mount('#__nuxt')
-
-    await nuxt.hooks.callHook('app:mounted', vueApp)
-    await nextTick()
+    try {
+      await applyPlugins(nuxt, plugins)
+    } catch (err) {
+      nuxt.hooks.hookOnce('app:error:handled', () => applyPlugins(nuxt, plugins))
+      await nuxt.callHook('app:error', err, nuxt)
```

Comment:
Hooks are already registered with same `nuxt` context. We don't need to pass it again.
```suggestion
      await nuxt.callHook('app:error', err)
```

Reply:
This was so the error handling hook receives nuxt to handle. Without this, it can't access `nuxt`.

Reply:
How then hook can be registered then? :D (`nuxt.hook('app:error', err => { /* we do already have access to `nuxt` })

Reply:
😆 Fair point!